src/kakarot/precompiles/kakarot_precompiles.cairo

```diff
@@ -148,7 +148,12 @@ namespace KakarotPrecompiles {
         // [to_address: address][data_offset: uint256][data_len: uint256][data: bytes[]]
 
         // Load target EVM address
-        let target_address = Helpers.bytes32_to_felt(input);
+        let invalid_address = Helpers.bytes_to_felt(12, input);
```

Comment:
do you think we can use bytes_used and check whether its ==20? if not, it's not a valid address and we revert. Perhaps it would be more efficient, although it's an isolated part of the codebase so the impact would be low

Reply:
So you mean in pseudo code:
```
let evm_address = Helpers.bytes_to_felt(32, input);
bytes_used = Helpers.bytes_used_256(evm_address)
if (bytes_used != 20) {
    <revert> 
}
```
We would need to create a new function `bytes_used_256` as with `bytes_used_128`, it would be necessary to split the felt and call  `bytes_used_128` on low and high which might not be more efficient.

A new issue could be created for `bytes_used_256` as I think it is encountered several times in the code base and could be done on the work that would be done in the optimisation issue of `bytes_used_128` https://github.com/kkrt-labs/kakarot/issues/1319

Reply:
hm, actually we just need to check if evm_address > 2**160 then we're sure it overflows the size of an EVM address, no need to know the actual bytes count.

Reply:
In fact I think it is not possible either: `let evm_address = Helpers.bytes_to_felt(32, input);` will also fail silently in case bytes do not fit in a felt  like the initial implementation. So the suggested alternative solution seems not viable.